aten/src/ATen/native/cuda/RNN.cu

```diff
@@ -0,0 +1,601 @@
+#include "ATen/ATen.h"
+#include "ATen/AccumulateType.h"
+#include "ATen/TensorUtils.h"
+#include "ATen/NativeFunctions.h"
+#include "ATen/cuda/CUDAContext.h"
+#include "ATen/cuda/CUDAApplyUtils.cuh"
+#include <THC/THCNumerics.cuh>
+
+namespace at { namespace native {
+
+namespace {
+
+using at::cuda::detail::TensorInfo;
+using at::cuda::detail::getTensorInfo;
+using at::cuda::detail::IndexToOffset;
+using at::cuda::detail::canUse32BitIndexMath;
+
+// Factor will be 3 for GRU and 4 for LSTM
+void checkSizes(CheckedFrom c,
+                const TensorArg& input_gates, const TensorArg& hidden_gates,
+                const TensorArg& input_bias, const TensorArg& hidden_bias,
+                int64_t factor, const TensorArg& prev_hidden) {
+  checkDim(c, input_gates, 2);
+  checkSameSize(c, input_gates, hidden_gates);
+  int64_t gates_size = input_gates->size(1);
+
+  if (input_bias->defined()) {
```

Comment:
Check `hidden_bias` is defined?

Reply:
It will fail in `checkSameSize` below. Do you want me to add an explicit check? Those functions are to be considered internal, so they don't need to be 100% bulletproof.

Reply:
If this is an internal check, it's OK.